src/main/java/jenkins/plugins/git/GitUsernamePasswordBinding.java

```diff
@@ -0,0 +1,208 @@
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+public class GitUsernamePasswordBinding extends MultiBinding<StandardUsernamePasswordCredentials> implements GitCredentialBindings {
+    final static private String GIT_USERNAME_KEY = "GIT_USERNAME";
+    final static private String GIT_PASSWORD_KEY = "GIT_PASSWORD";
+    final private String gitToolName;
+    private transient boolean unixNodeType;
+
+    @DataBoundConstructor
+    public GitUsernamePasswordBinding(String gitToolName, String credentialsId) {
+        super(credentialsId);
+        this.gitToolName = gitToolName;
+        //Variables could be added if needed
+    }
+
+    public String getGitToolName(){
+        return this.gitToolName;
+    }
+
+    private void setUnixNodeType(boolean value) {
+        this.unixNodeType = value;
+    }
+
+    @Override
+    protected Class<StandardUsernamePasswordCredentials> type() {
+        return StandardUsernamePasswordCredentials.class;
+    }
+
+    @Override
+    public MultiEnvironment bind(@NonNull Run<?, ?> run, FilePath filePath,
+                                 Launcher launcher, @NonNull TaskListener taskListener)
+            throws IOException, InterruptedException {
+        final Map<String, String> secretValues = new LinkedHashMap<>();
+        final Map<String, String> publicValues = new LinkedHashMap<>();
+        StandardUsernamePasswordCredentials credentials = getCredentials(run);
+        setCredentialPairBindings(credentials,secretValues,publicValues);
+        GitTool cliGitTool = getCliGitTool(run, this.gitToolName, taskListener);
+        if (cliGitTool != null && filePath != null) {
+            final UnbindableDir unbindTempDir = UnbindableDir.create(filePath);
+            setUnixNodeType(isCurrentNodeOSUnix(launcher));
+            setGitEnvironmentVariables(getGitClientInstance(cliGitTool.getGitExe(), unbindTempDir.getDirPath(),
+                                                            new EnvVars(), taskListener), publicValues);
+            GenerateGitScript gitScript = new GenerateGitScript(
+                                                  credentials.getUsername(),
+                                                  credentials.getPassword().getPlainText(),
+                                                  credentials.getId(),
+                                                  this.unixNodeType);
+            FilePath gitTempFile = gitScript.write(credentials, unbindTempDir.getDirPath());
+            secretValues.put("GIT_ASKPASS", gitTempFile.getRemote());
+            return new MultiEnvironment(secretValues, publicValues, unbindTempDir.getUnbinder());
+        } else {
+            taskListener.getLogger().println("JGit and JGitApache type Git tools are not supported by this binding");
```

Comment:
`getCliGitTool` could return a null value for a lot of other reasons than just the possibility of git tool being JGit and JGitApache types. I suggest you should make this log statement more general: 

"Unable to perform binding because git tool resolution has failed"